src/Psalm/Checker/MethodChecker.php

```diff
@@ -660,4 +661,35 @@ public static function compareMethods(
             return null;
         }
     }
+
+    /**
+     * @param  MethodStorage $method_storage
+     * @param  CodeLocation  $code_location
+     * @return false|null
+     */
+    public static function checkMethodSignatureMustOmitReturnType(
+        MethodStorage $method_storage,
+        CodeLocation $code_location
+    ) {
+        if ($method_storage->signature_return_type === null) {
+            return null;
+        }
+
+        $cased_method_name = $method_storage->cased_name;
+        switch ($cased_method_name) {
+        case '__clone':
```

Comment:
Is this a common pattern? I'm used to indenting `case` statements

Reply:
Don't know. I can change it to a series of `===`/`||` or an `in_array` check instead.

Reply:
no I just meant I'm used to seeing/writing

```php
switch ($foo) {
    case 'bar':
        // do something
}
```

Reply:
Non-indented case is pretty common, yes. But from a quick look at the code base I see that it indents them everywhere.